tests/unit/Core/Content/Product/SalesChannel/Price/ProductPriceCalculatorTest.php

```diff
@@ -83,6 +83,8 @@ public function testPriceWillBeCalculated(Entity $entity, ?PriceAssertion $expec
         static::assertEquals($expected->reference, $price->getReferencePrice()?->getPrice());
 
         static::assertEquals($expected->listPrice, $price->getListPrice()?->getPrice());
+
+        static::assertEquals($expected->regulation, $price->getRegulationPrice()?->getPrice());
```

Comment:
please use `assertSame` where possible instead, as it is type safe

Reply:
solved, we can probably enforce it by a phpstan rule? not sure in which case `assertEquals` is acceptable and unavoidable